### PR TITLE
test uniq validation empty attribute

### PR DESCRIPTION
**Pull request by yamaha252**
error unique validate if empty attribute

### CODE REVIEW

File: tests/unit/framework/validators/UniqueValidatorTest.php [hunk 1]
```diff
@@ -136,4 +136,17 @@ public function testValidateCompositeKeys()
...
+        $val = new UniqueValidator();
+
+        $m = new ValidatorTestRefModel(['ref' => '']);
+        $val->validateAttribute($m, 'ref');
+        $this->assertFalse($m->hasErrors('ref'));
+        $m->save(false);
+
+        $m = new ValidatorTestRefModel(['ref' => '']);
+        $val->validateAttribute($m, 'ref');
+        $this->assertTrue($m->hasErrors('ref'));
```
Comment (cebe) [inline tests/unit/framework/validators/UniqueValidatorTest.php]: why do you expect it to have errors here?

Comment (yamaha252) [inline tests/unit/framework/validators/UniqueValidatorTest.php]: here is added new record which contains duplicate empty string in `ref` field

**Comment by yamaha252** [comment]
after save an empty string is converted to null in database, and validator created a query with `string ""`

File: tests/unit/framework/validators/UniqueValidatorTest.php [hunk 2]
```diff
@@ -136,4 +136,30 @@ public function testValidateCompositeKeys()
...
+        $val = new UniqueValidator();
+
+        $m = new ValidatorTestMainModel(['field1' => '']);
+        $val->validateAttribute($m, 'field1');
+        $this->assertFalse($m->hasErrors('field1'));
+        $m->save(false);
+
+        $m = new ValidatorTestMainModel(['field1' => '']);
+        $val->validateAttribute($m, 'field1');
+        $this->assertTrue($m->hasErrors('field1'));
```
Comment (yamaha252) [inline tests/unit/framework/validators/UniqueValidatorTest.php]: correct working assert in this

**Comment by cebe** [comment]
That is not the problem, if you insert a string into a integer column, that does not work. You should ensure the column value is int by adding integer validator or filter validator. Merged your tests with slight adjustment. Thanks!

**Pull request closed without merging**